unblob/handlers/filesystem/ubi.py

```diff
@@ -137,8 +137,14 @@ def _walk_ubi(self, file: File, peb_size: int) -> int:
             first_bytes = file.read(len(self._UBI_EC_HEADER))
             if first_bytes == b"" or first_bytes != self._UBI_EC_HEADER:
                 break
-            file.seek(offset + peb_size)
-
+            try:
+                file.seek(offset + peb_size)
+            except SeekError:
+                last_chunk_size = file.size() - file.tell() - len(first_bytes)
+                # if aligned, we accept to go to the end even if not full PEB
+                if peb_size % last_chunk_size == 0:
```

Comment:
I do not understand this alignment check, what needs to be aligned here?

we only check that the peb_size must be the multiple of the last_chunk_size, but does it make the last chunk size valid?

eg: peb size is 256k, then even a 8 or 16byte last chunk would make it valid, what are we really checking here?

Reply:
I got lost trying to make it work with a very specific sample so as not to have an unknown chunk at the end. Please forget about this :)

The only thing we need is to break on `SeekError`, which is what we do now.